### ISSUE TITLE
[RyuJIT/arm32] Assert failure: pImportSection == pModule->GetImportSectionForRVA(rva)

### ISSUE THREAD

**Issue by BruceForstall** [dotnet/coreclr#15150]
Test failure with Checked ARM run on Release test build, on Windows hardware.

Happens with JitStressRegs=1 and JitStressRegs=8.

https://ci.dot.net/job/dotnet_coreclr/job/master/job/jitstress/job/arm_cross_checked_windows_nt_jitstressregs1_tst_prtest/2/artifact/bin/tests/Windows_NT.arm.Checked/Smarty.run.0/Smarty.0.fail.smrt/*view*/
https://ci.dot.net/job/dotnet_coreclr/job/master/job/jitstress/job/arm_cross_checked_windows_nt_jitstressregs8_tst_prtest/2/artifact/bin/tests/Windows_NT.arm.Checked/Smarty.run.0/Smarty.0.fail.smrt/*view*/

Similar (or identical) bugs: #12659, #12660

Tests:
```
readytorun\tests\mainv1\mainv1.cmd
readytorun\tests\mainv2\mainv2.cmd
```

Message:

```
Assert failure(PID 18548 [0x00004874], Thread: 20360 [0x4f88]): pImportSection == pModule->GetImportSectionForRVA(rva)

<no module>! <no symbol> + 0x0 (0x00000000)
    File: e:\gh\coreclr2\src\vm\prestub.cpp Line: 2090
    Image: c:\brucefo\core_root\Windows_NT.arm.Checked.2017-11-11\CoreRun.exe
```

**Comment by BruceForstall** [comment]
@dotnet/arm32-contrib

**Comment by BruceForstall** [comment]
Also:

https://ci.dot.net/job/dotnet_coreclr/job/master/job/jitstress/job/arm_cross_checked_windows_nt_jitstress2_jitstressregs1_tst_prtest/2/

**Comment by alpencolt** [comment]
There is the same error on Linux.
Could you assign this issue to me?

**Comment by BruceForstall** [comment]
@alpencolt Have you had time to look at this?

**Comment by alpencolt** [comment]
@BruceForstall little bit. It looks something wrong in crossgen...
I do it on background and will not finish this week and unfortunately next too. I don't mind if someone will start to work on this issue too but please write about it;)

**Comment by alpencolt** [comment]
`test.ni.dll` generated in stress mode incorrect. `GetImportSectionForRVA(rva)` returns `NULL`.

**Comment by sandreenko** [comment]
Some r2r tests hit the same issue without stress mode:
JIT\Methodical\Arrays\lcs\_speed_dbglcsbox\_speed_dbglcsbox.cmd 
JIT\Directed\StrAccess\straccess2_cs_ro\straccess2_cs_ro.cmd 
JIT\Performance\CodeQuality\Roslyn\CscBench\CscBench.cmd 
JIT\Methodical\Arrays\lcs\_speed_rellcsbox\_speed_rellcsbox.cmd 
JIT\Performance\CodeQuality\Serialization\Deserialize\Deserialize.cmd 
JIT\Performance\CodeQuality\Serialization\Serialize\Serialize.cmd 
JIT\Performance\CodeQuality\BenchmarksGame\fasta\fasta-1\fasta-1.cmd 
managed\Compilation\Compilation\Compilation.cmd 

will be fixed as part of #15552

**Comment by sandreenko** [comment]
Fixed via #15910.

**Comment by BruceForstall** [comment]
#15910 was reverted (#16218)
